DOCSP-30559: compatibility (#3)
mongodb/docs-rust: source/index.txt

Compatibility
-------------

> To see the charts that show the recommended {+driver-short+} version
> for each MongoDB Server version and the compatible language versions for
> each driver version, see :ref:`rust-compatibility`.

Reviewer: I: This sentence is a little hard to parse (and contains the word "version" four times, which might be a record). Removed "recommended" because I don't think that page actually recommends anything.

```suggestion
To see the versions of MongoDB Server and Rust that are compatible with
each version of the {+driver-short+}, see :ref:`rust-compatibility`.
```

Reviewer: The above has "see" twice, which I don't love. Could also do this:

```suggestion
For more information about the {+driver-short+}'s compatibility with 
different versions of MongoDB Server and the Rust language, see
:ref:`rust-compatibility`.
```

Author: good call
